### ISSUE TITLE
Objects should have a more robust identifer other than name

### ISSUE THREAD

**Issue by jgostick** [PMEAL/OpenPNM#1594]
``obj.name`` is handy and user friendly, but it is actually a bit fragile, and if for some reason the object loses is ``project`` the whole system breaks.  I think the back end should be re-factored to look for uuids instead of names.  These never change and are never in conflict. 

Given the above plan it would also be possible to keep a handle for every single object in the ``Workspace`` , indexed by uuid.  This would be a master list of objects, and could be used to query global info like which project they belong to, etc.  

Basically, I'm finding that the Workspace/Project system needs work.  :-(
